File: backend/utils/response.py
class Response:

    def __init__(self, responseType, data=None):
        self.type = responseType
        self.data = data

    def __str__(self):
        return "{}: {}".format(self.type, self.data)


class SignUpOkResponse(Response):
    def __init__(self):
        super().__init__("SIGN_UP_OK")


class SignUpFailedResponse(Response):
    def __init__(self):
        super().__init__("SIGN_UP_FAILED")


class DuplicateSignUpEmailResponse(Response):
    def __init__(self):
        super().__init__("DUPLICATE_SIGN_UP_EMAIL")


class SignInOkResponse(Response):
    def __init__(self, data):
        super().__init__("SIGN_IN_OK", data)


class SignInFailedResponse(Response):
    def __init__(self):
        super().__init__("SIGN_IN_FAILED")


class FoundPersonResponse(Response):
    def __init__(self, data):
        super().__init__("FOUND_PERSON", data)


class UnknownPersonResponse(Response):
    def __init__(self):
        super().__init__("UNKNOWN_PERSON")


class NoFaceDetectedResponse(Response):
    def __init__(self):
        super().__init__("NO_FACE_DETECTED")

File: frontend/lib/utils/response.dart
class Response {
  final String type;
  final String? data;

  Response(this.type, this.data);
  @override
  String toString() {
    return '$type: $data';
  }
}

const String SIGN_UP_OK = "SIGN_UP_OK";
const String SIGN_UP_FAILED = "SIGN_UP_FAILED";
const String DUPLICATE_SIGN_UP_EMAIL = "DUPLICATE_SIGN_UP_EMAIL";
const String SIGN_IN_OK = "SIGN_IN_OK";
const String SIGN_IN_FAILED = "SIGN_IN_FAILED";
const String FOUND_PERSON = "FOUND_PERSON";
const String UNKNOWN_PERSON = "UNKNOWN_PERSON";
const String NO_FACE_DETECTED = "NO_FACE_DETECTED";
